app/models/downloaded_file.rb

```diff
@@ -0,0 +1,30 @@
+require 'rmt/checksum_verifier'
+
+class DownloadedFile < ApplicationRecord
+
+  def self.add_file!(checksum_type, checksum, local_path)
+    return nil unless local_path.match?(/\.(rpm|drpm)$/)
```

Comment:
That might not be the best logic:

In `RMT::Mirror` there are `mirror_metadata` and `mirror_license` methods, which download all of the metadata. In those methods all the files are known to be metadata and those files we don't want to deduplicate.

`mirror_data` method parses `primary.xml` file of the repo and downloads files that are mentioned there. However, it can reference files with arbitrary names (and extensions). While most commonly in the modern SUSE repos the extensions are `.rpm` and `.drpm`, that doesn't necessarily apply for older or third-party repos.

I would suggest moving all of the deduplication method calls out of `RMT::Downloader` class into `mirror_data` method `RMT::Mirror` class.